src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -678,10 +678,48 @@ namespace winrt::TerminalApp::implementation
         if (_tabs.Size() == 0 && !(_shouldStartInboundListener || _isEmbeddingInboundListener))
         {
             _LastTabClosedHandlers(*this, nullptr);
+            co_return;
         }
         else
         {
-            _InitializedHandlers(*this, nullptr);
+            // GH#11561: When we start up, our window is initially just a frame
+            // with a transparent content area. We're gonna do all this startup
+            // init on the UI thread, so the UI won't actually paint till it's
+            // all done. This results in a few frames where the frame is
+            // visible, before the page paints for the first time, before any
+            // tabs appears, etc.
+            //
+            // To mitigate this, we're gonna wait for the UI thread to finish
+            // everything it's gotta do for the initial init, and _then_ fire
+            // our Initialized event. By waiting for everything else to finish
+            // (CoreDispatcherPriority::Low), we let all the tabs and panes
+            // actually get created. In the window layer, we're gonna cloak the
+            // window till this event is fired, so we don't actually see this
+            // frame until we're actually all ready to go.
+            //
+            // This will result in the window seemingly not loading as fast, but
+            // it will actually take exactly the same amount of time before it's
+            // usable.
+            //
+            // We also experimented with drawing a solid BG color before the
+            // initialization is finished. However, there are still a few frames
+            // after the frame is displayed before the XAML content first draws,
+            // so that didn't actually resolve any issues.
+
+            // Capture calling context.
+            winrt::apartment_context ui_thread;
```

Comment:
Unused?

Reply:
yea - I wanted to just `co_await ui_thread`, but that doesn't let me await it with low priority 🤷

Reply:
We should try to avoid using `winrt::apartment_context` as it's inherently flawed unfortunately. 😥
As Raymond Chen wrote:
> Unfortunately, it's true. co_await'ing an `apartment_context` is blocking [the background thread]. That's because it's built out of COM's `IContextCallback`, [which supports only synchronous transfer](https://devblogs.microsoft.com/oldnewthing/20191128-00/?p=103157).
> [...]
> Note that it's even worse: Even after you get to the other side, the side you came **\*from\*** is still blocked until you finish doing your other-side work (which could take the form of another co_await).
> [...]
> If the UI thread is hung, this will lead to threadpool starvation and eventually a deadlock.